.github/workflows/e2e-arm64.yaml

```diff
@@ -23,11 +23,12 @@ jobs:
     - env:
         TARGET: ${{ matrix.target }}
       run: |
+        set -euo pipefail
+
         echo "${TARGET}"
         case "${TARGET}" in
           linux-arm64-e2e)
-            PASSES='build release e2e' CPU='4' EXPECT_DEBUG='true' COVER='false' RACE='true' ./scripts/test.sh 2>&1 | tee test.log
-            ! grep -E "(--- FAIL:|FAIL:|DATA RACE|panic: test timed out|appears to have leaked)" -B50 -A10 test.log
+            PASSES='build release e2e' CPU='4' EXPECT_DEBUG='true' COVER='false' RACE='true' ./scripts/test.sh
```

Comment:
Does this affect child bash that is created when running scripts?

Reply:
Good point! Unfortunately, child bash doesn't inherit it. 

```bash
➜  ~ bash
fuwei@wenchang:~$ echo $SHELLOPTS
braceexpand:emacs:hashall:histexpand:history:interactive-comments:monitor
fuwei@wenchang:~$
fuwei@wenchang:~$ # no pipefail
fuwei@wenchang:~$ (echo 123; exit 2) | tee
123
fuwei@wenchang:~$ echo $?
0
fuwei@wenchang:~$
fuwei@wenchang:~$ set -o pipefail
fuwei@wenchang:~$ echo $SHELLOPTS
braceexpand:emacs:hashall:histexpand:history:interactive-comments:monitor:pipefail
fuwei@wenchang:~$
fuwei@wenchang:~$ (echo 123; exit 2) | tee
123
fuwei@wenchang:~$ echo $?
2
fuwei@wenchang:~$
fuwei@wenchang:~$ # run a child bash to check inherit
fuwei@wenchang:~$ cat /tmp/testing.sh
#!/usr/bin/env bash

echo $SHELLOPTS

(echo 12345; exit 2) | tee
fuwei@wenchang:~$ /tmp/testing.sh
braceexpand:hashall:interactive-comments
12345
fuwei@wenchang:~$ echo $?
0
fuwei@wenchang:~$
fuwei@wenchang:~$ export SHELLOPTS
fuwei@wenchang:~$ # run again
fuwei@wenchang:~$ /tmp/testing.sh
braceexpand:emacs:hashall:histexpand:history:interactive-comments:monitor:pipefail
12345
fuwei@wenchang:~$ echo $?
2
fuwei@wenchang:~$
```


The previous test https://github.com/etcd-io/etcd/actions/runs/4453530207/jobs/7822105485 pass because the `test.sh` sets `set -e` and `pipefail` works in parent side. I think I should export `$SHELLOPTS` to ensure all the bash run with the setting. 

> SHELLOPTS
A colon-separated list of enabled shell options. Each word in the list is a valid argument for the -o option to the set builtin command (see [The Set Builtin](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html)). The options appearing in SHELLOPTS are those reported as ‘on’ by ‘set -o’. If this variable is in the environment when Bash starts up, each shell option in the list will be enabled before reading any startup files. This variable is readonly.

Does it make senses?

REF: https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#index-SHELLOPTS

Reply:
Adding `SHELLOPTS` will not work for local development. Could we start from adding `set -euo pipefail` to `./scripts/test.sh`?

Reply:
OK. Will update it.